src/lib/common/Settings.cpp

```diff
@@ -53,10 +68,17 @@ Settings::Settings(QObject *parent) : QObject(parent)
   else
     fileToLoad = UserSettingFile;
 
-  m_settings = new QSettings(fileToLoad, QSettings::IniFormat);
+  m_settings = new QSettings(fileToLoad, QSettings::IniFormat, this);
   m_settingsProxy = std::make_shared<QSettingsProxy>();
   m_settingsProxy->load(fileToLoad);
   qInfo().noquote() << "initial settings file:" << m_settings->fileName();
+
+  const auto stateBase = !qEnvironmentVariable("XDG_STATE_HOME").isEmpty()
+                             ? qEnvironmentVariable("XDG_STATE_HOME")
+                             : QStandardPaths::standardLocations(QStandardPaths::GenericStateLocation).at(0);
```

Comment:
This would be easier to read if `qEnvironmentVariable("XDG_STATE_HOME")` was encapsulated into a var.
```suggestion
  const auto xdgStateHome = qEnvironmentVariable("XDG_STATE_HOME");
  const auto stateBase = !xdgStateHome.isEmpty()
                             ? xdgStateHome
                             : QStandardPaths::standardLocations(QStandardPaths::GenericStateLocation).at(0);
```

Reply:
Added this in a new commit where we do this or both the env vars used in this method.